forge-game/src/main/java/forge/game/ability/effects/AnimateEffectBase.java

```diff
@@ -79,16 +84,38 @@ public static void doAnimate(final Card c, final SpellAbility sa, final Integer
         }
 
         if (!addType.isEmpty() || !removeType.isEmpty() || addAllCreatureTypes || !remove.isEmpty()) {
```

Comment:
Right now this check always passes, so if perpetual = true it builds a useless Pair

I think it's because addType always isn't empty because of the incomplete CardType?

Reply:
Is it the `!` that should be removed?
<img width="580" alt="image" src="https://github.com/Card-Forge/forge/assets/103371817/71f32cc6-4d46-4b29-90ae-135e617337b6">

Reply:
yea I think so

Reply:
Upon removing the `!` it is behaving more as I would expect it to 😁